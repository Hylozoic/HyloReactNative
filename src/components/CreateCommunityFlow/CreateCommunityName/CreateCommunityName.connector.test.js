import {
  mapStateToProps,
  mapDispatchToProps,
  mergeProps
} from './CreateCommunityName.connector'

const communityName = 'name'

const state = {
  CreateCommunityFlow: {
    communityName
  }
}

describe('mapStateToProps', () => {
  it('returns the right keys', () => {
    expect(mapStateToProps(state, {})).toMatchSnapshot()
  })
})

describe('mapDispatchToProps', () => {
  it('maps the action generators', () => {
    expect(mapDispatchToProps.hasOwnProperty('saveCommunityName')).toBeTruthy()
    expect(mapDispatchToProps.saveCommunityName(communityName)).toMatchSnapshot()
  })
})

describe('mergeProps', () => {
  it('merges the props', () => {
    const ownProps = {
      navigation: {
        navigate: jest.fn()
      }
    }
    const mergedProps = mergeProps(state, {}, ownProps)
    expect(mergedProps).toMatchSnapshot()
    mergedProps.goToCreateCommunityUrl()
    expect(ownProps.navigation.navigate).toHaveBeenCalled()
  })
})
