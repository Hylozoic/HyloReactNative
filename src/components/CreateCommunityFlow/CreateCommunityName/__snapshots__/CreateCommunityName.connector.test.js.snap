// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapDispatchToProps maps the action generators 1`] = `
Object {
  "payload": "name",
  "type": "CreateCommunityFlow/SAVE_COMMUNITY_NAME",
}
`;

exports[`mapStateToProps returns the right keys 1`] = `
Object {
  "communityName": "name",
}
`;

exports[`mergeProps merges the props 1`] = `
Object {
  "CreateCommunityFlow": Object {
    "communityName": "name",
  },
  "goToCreateCommunityUrl": [Function],
  "navigation": Object {
    "navigate": [Function],
  },
  "saveCommunityName": [Function],
}
`;
